src/tools/rbd_mirror/image_map/Types.h

```diff
@@ -0,0 +1,116 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_RBD_MIRROR_IMAGE_MAP_TYPES_H
+#define CEPH_RBD_MIRROR_IMAGE_MAP_TYPES_H
+
+#include <set>
+#include <map>
+#include <string>
+#include <memory>
+
+#include "cls/rbd/cls_rbd_types.h"
+
+namespace rbd {
+namespace mirror {
+namespace image_map {
+
+enum ShuffleType {
+  SHUFFLE_TYPE_INSTANCES_ADDED = 0,   // rebalance due to addition of instances
+  SHUFFLE_TYPE_INSTANCES_REMOVED,     // rebalance due to removal of instances
+  SHUFFLE_TYPE_IMAGES_ADDED_REMOVED,  // rebalance due to adding/removing images
+};
+
+struct Image {
+  // to_instance_id is unused as of now and always
+  // matches instance_id. When we have rebalancer
+  // in place to_instance_id would point to the
+  // instance id of the mirror daemon where the
+  // image needs to be reassigned (from currently
+  // instance_id).
```

Comment:
Nit: I would add a state to this so you can track (in-memory) your current and next actions in one place instead of having multiple lists of things to map / unmap / waiting for instance response / etc.